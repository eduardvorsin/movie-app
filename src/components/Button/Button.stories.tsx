import type { Meta, StoryObj } from '@storybook/react';
import Button, { Props } from './Button';
import Add from '../../assets/icons/add.svg';
import Minus from '../../assets/icons/minus.svg';

type Sizes = Exclude<Props['size'], undefined>;

const sizes: Sizes[] = ['micro', 'slim', 'medium', 'large'];

type Variant = {
	appearance: Props['appearance'],
	children: string,
	isLoading?: Props['isLoading'],
	isDisabled?: Props['isDisabled'],
	hasIcon?: boolean,
};

type AllVariants = {
	micro: Variant[],
	slim: Variant[],
	medium: Variant[],
	large: Variant[],
	fullWidth: Variant[],
};

const all: AllVariants = {
	micro: [
		{
			appearance: 'primary',
			children: 'micro primary',
			isLoading: false,
		},
		{
			appearance: 'secondary',
			children: 'micro secondary',
		},
		{
			appearance: 'warning',
			children: 'micro warning',
		},
		{
			appearance: 'danger',
			children: 'micro danger',
		},
		{
			appearance: 'primary',
			children: 'micro loading',
			isLoading: true,
		},
		{
			appearance: 'secondary',
			children: 'micro disabled',
			isDisabled: true,
		},
		{
			appearance: 'warning',
			children: 'micro icon',
			hasIcon: true,
		}
	],
	slim: [
		{
			appearance: 'primary',
			children: 'slim primary',
		}, {
			appearance: 'secondary',
			children: 'slim secondary',
		},
		{
			appearance: 'warning',
			children: 'slim warning',
		},
		{
			appearance: 'danger',
			children: 'slim danger',
		},
		{
			appearance: 'primary',
			children: 'slim loading',
			isLoading: true,
		},
		{
			appearance: 'secondary',
			children: 'slim disabled',
			isDisabled: true,
		},
		{
			appearance: 'warning',
			children: 'slim icon',
			hasIcon: true,
		}
	],
	medium: [
		{
			appearance: 'primary',
			children: 'medium primary',
		}, {
			appearance: 'secondary',
			children: 'medium secondary',
		},
		{
			appearance: 'warning',
			children: 'medium warning',
		},
		{
			appearance: 'danger',
			children: 'medium danger',
		},
		{
			appearance: 'primary',
			children: 'medium loading',
			isLoading: true,
		},
		{
			appearance: 'secondary',
			children: 'medium disabled',
			isDisabled: true,
		},
		{
			appearance: 'warning',
			children: 'medium icon',
			hasIcon: true,
		}
	],
	large: [
		{
			appearance: 'primary',
			children: 'large primary',
		}, {
			appearance: 'secondary',
			children: 'large secondary',
		},
		{
			appearance: 'warning',
			children: 'large warning',
		},
		{
			appearance: 'danger',
			children: 'large danger',
		},
		{
			appearance: 'primary',
			children: 'large loading',
			isLoading: true,
		},
		{
			appearance: 'secondary',
			children: 'large disabled',
			isDisabled: true,
		},
		{
			appearance: 'warning',
			children: 'large icon',
			hasIcon: true,
		}
	],
	fullWidth: [
		{
			appearance: 'primary',
			children: 'full width primary',
		}, {
			appearance: 'secondary',
			children: 'full width secondary',
		},
		{
			appearance: 'warning',
			children: 'full width warning',
		},
		{
			appearance: 'danger',
			children: 'full width danger',
		},
		{
			appearance: 'primary',
			children: 'full width loading',
			isLoading: true,
		},
		{
			appearance: 'secondary',
			children: 'full width disabled',
			isDisabled: true,
		},
		{
			appearance: 'warning',
			children: 'full width icon',
			hasIcon: true,
		}
	],
};

const meta: Meta<typeof Button> = {
	title: 'components/Button',
	tags: ['autodocs'],
	argTypes: {
		appearance: {
			description: 'Button appearance option',
		},
		className: {
			description: 'Additional classes for the button'
		},
		isDisabled: {
			description: 'Disabling the button makes it non-functional',
		},
		isLoading: {
			description: 'Substitutes the button text with a loading indicator and deactivates the button',
		},
		fullWidth: {
			description: 'The button spans the entire width of its parent container',
		},
		children: {
			description: 'The button\'s contents',
		},
		size: {
			description: 'Sets the size of the button. You can choose a ready-made or choose a custom option and set your own styles for the sizes.',
		},
		iconBefore: {
			description: 'The icon that will be placed before the content of the button',
		},
		iconAfter: {
			description: 'The icon that will be placed after the content of the button',
		},
		onBlur: {
			control: false,
			description: 'The callback is executed when the button loses focus'
		},
		onFocus: {
			control: false,
			description: 'The callback is executed when the button gets focus'
		},
		onClick: {
			control: false,
			description: 'The callback is performed when the button is pressed'
		},
		testId: {
			description: 'Id for testing the component',
		}
	},
	component: Button,
	parameters: {
		docs: {
			description: {
				component: 'Buttons serve as initiators for various actions. They find utility within forms, toolbars, modal dialogs, and even function as independent action triggers. These actions predominantly take place within the confines of the same page.',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof Button>;

export const Primary: Story = {
	args: {
		appearance: 'primary',
		children: 'Primary'
	},
	parameters: {
		docs: {
			description: {
				story: 'Applies to buttons with the primary importance of the action',
			},
		},
	},
};

export const Secondary: Story = {
	args: {
		appearance: 'secondary',
		children: 'Secondary'
	},
	parameters: {
		docs: {
			description: {
				story: 'Used for buttons that perform a secondary action',
			},
		},
	},
};

export const Warning: Story = {
	args: {
		appearance: 'warning',
		children: 'Warning'
	},
	parameters: {
		docs: {
			description: {
				story: 'It is used to warn about the potential risk of interaction',
			},
		},
	},
};

export const Danger: Story = {
	args: {
		appearance: 'danger',
		children: 'Danger'
	},
	parameters: {
		docs: {
			description: {
				story: 'It is used to warn the user about the irrevocability of actions when clicking',
			},
		},
	},
};

export const Loading: Story = {
	args: {
		isLoading: true,
		children: 'Loading'
	},
	parameters: {
		docs: {
			description: {
				story: 'Enabling the isLoading property as true will substitute the button text with a loading indicator and deactivate the button to avoid unintended submissions. The button\'s width will remain consistent to prevent any content from shifting.',
			},
		},
	},
};

export const Disabled: Story = {
	args: {
		isDisabled: true,
		children: 'Disabled'
	},
	parameters: {
		docs: {
			description: {
				story: 'Apply this for actions that are presently unavailable, like right after submitting a form. The user interface should effectively convey why the button is inactive and what steps are necessary to activate it.',
			},
		},
	},
};

export const FullWidth: Story = {
	args: {
		fullWidth: true,
		children: 'Full Width'
	},
	parameters: {
		docs: {
			description: {
				story: 'Use the fullWidth property to allow the button to expand to the full width of its container',
			},
		},
	},
};

export const EventCallbacks: Story = {
	argTypes: {
		onClick: {
			action: 'Clicked',
		},
		onFocus: {
			action: 'Focused',
		},
		onBlur: {
			action: 'Blured',
		}
	},
	args: {
		fullWidth: true,
		children: 'click, focus, blur events'
	},
	parameters: {
		docs: {
			description: {
				story: 'Callback functions can be supplied to onClick, onBlur, and onFocus events',
			},
		},
	},
};

export const WithIcons: Story = {
	parameters: {
		docs: {
			description: {
				story: 'Through the iconBefore and IconAfter properties, you can add icons to the button',
			},
		},
	},
	render: () => (
		<div
			className='flex items-start'
		>
			<Button
				className='mr-3'
				iconBefore={<Add />}
			>
				Left side icon
			</Button>
			<Button
				className='mr-3'
				iconAfter={<Minus />}
			>
				Right side icon
			</Button>
			<Button
				iconBefore={<Add />}
				iconAfter={<Minus />}
			>
				Both side icons
			</Button>
		</div>
	)
};

export const Sizes: Story = {
	render: () => (
		<div className='flex items-start'>
			{sizes.map((size) => (
				<Button
					className='mr-3 last:mr-0'
					key={size}
					size={size}
				>
					{`${size[0].toUpperCase()}${size.slice(1)}`}
				</Button>
			))}
		</div>
	),
	parameters: {
		docs: {
			description: {
				story: 'Allows you to set the size of the button. The default size is medium. There are 5 sizes: micro, slim, medium, large, custom. The first 4 sizes have an explicit size and custom allows you to set your own sizes through css classes',
			},
		},
	},
};

export const All: Story = {
	render: () => {
		return (
			<div className='grid gap-y-6 items-start'>
				{(Object.keys(all) as Array<keyof typeof all>).map((size) => (
					<div
						key={size}
						className='flex flex-wrap items-start'
					>
						{all[size].map((variant) => (
							<Button
								className='mr-3 last:mr-0 mb-3 last:mb-0'
								key={variant.children}
								size={size === 'fullWidth' ? 'medium' : size}
								appearance={variant.appearance}
								isDisabled={variant.isDisabled}
								isLoading={variant.isLoading}
								fullWidth={size === 'fullWidth'}
								iconBefore={variant.hasIcon ? <Add /> : undefined}
								iconAfter={variant.hasIcon ? <Minus /> : undefined}
							>
								{variant.children}
							</Button>
						))}
					</div>
				))}
			</div>
		)
	},
	parameters: {
		docs: {
			description: {
				story: 'Here are all possible button options, in size, appearance and condition',
			},
		},
	},
};

