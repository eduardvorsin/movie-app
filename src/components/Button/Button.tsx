'use client';
import { FocusEventHandler, MouseEventHandler, ReactNode } from 'react';
import Spinner from '@/components/Spinner/Spinner';
import Link from 'next/link';

export type Props = {
	appearance?: 'primary' | 'secondary' | 'warning' | 'danger' | 'success' | 'discovery' | 'ghost',
	className?: string,
	iconAfter?: ReactNode,
	iconBefore?: ReactNode,
	isDisabled?: boolean,
	isLoading?: boolean,
	fullWidth?: boolean,
	onBlur?: FocusEventHandler<HTMLElement>,
	onClick?: MouseEventHandler<HTMLElement>,
	onFocus?: FocusEventHandler<HTMLElement>,
	size?: 'micro' | 'slim' | 'medium' | 'large' | 'custom',
	children: ReactNode,
	testId?: string,
	href?: string,
	iconButton?: boolean,
};

const appearanceTypes = {
	primary: 'bg-blue-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-blue-800 enabled:active:bg-blue-900 dark:bg-blue-400 dark:enabled:hover:bg-blue-300 dark:enabled:active:bg-blue-200',
	secondary: 'bg-neutral-300 text-dark-neutral-0 enabled:hover:bg-neutral-400 enabled:active:bg-neutral-500 dark:enabled:hover:bg-neutral-200 dark:enabled:active:bg-neutral-100',
	success: 'bg-green-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-green-800 enabled:active:bg-green-900 dark:bg-green-400 dark:enabled:hover:bg-green-300 dark:enabled:active:bg-green-200',
	discovery: 'bg-purple-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-purple-800 enabled:active:bg-purple-900 dark:bg-purple-400 dark:enabled:hover:bg-purple-300 dark:enabled:active:bg-purple-200',
	danger: 'bg-red-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-red-800 enabled:active:bg-red-900 dark:bg-red-400 dark:enabled:hover:bg-red-300 dark:enabled:active:bg-red-200',
	warning: 'bg-yellow-500 text-neutral-1000 enabled:hover:bg-yellow-600 enabled:active:bg-yellow-700 dark:enabled:hover:bg-yellow-300 dark:enabled:active:bg-yellow-200',
	ghost: 'text-dark-neutral-0 enabled:hover:text-dark-neutral-400 enabled:active:text-dark-neutral-600 dark:text-neutral-200 dark:enabled:hover:text-neutral-400 dark:enabled:active:text-neutral-600',
} as const;

const iconButtonAppearances = {
	primary: 'text-blue-700 enabled:hover:text-blue-800 enabled:active:text-blue-900 dark:text-blue-400 dark:enabled:hover:text-blue-300 dark:enabled:active:text-blue-200',
	secondary: 'text-neutral-300 enabled:hover:text-neutral-400 enabled:active:text-neutral-500 dark:enabled:hover:text-neutral-200 dark:enabled:active:text-neutral-100',
	success: 'text-green-700 enabled:hover:text-green-800 enabled:active:text-green-900 dark:text-green-400 dark:enabled:hover:text-green-300 dark:enabled:active:text-green-200',
	discovery: 'text-purple-700 enabled:hover:text-purple-800 enabled:active:text-purple-900 dark:text-purple-400 dark:enabled:hover:text-purple-300 dark:enabled:active:text-purple-200',
	danger: 'text-red-700 enabled:hover:text-red-800 enabled:active:text-red-900 dark:text-red-400 dark:enabled:hover:text-red-300 dark:enabled:active:text-red-200',
	warning: 'text-yellow-500 enabled:hover:text-yellow-600 enabled:active:text-yellow-700 dark:enabled:hover:text-yellow-300 dark:enabled:active:text-yellow-300',
	ghost: 'text-dark-neutral-0 enabled:hover:text-dark-neutral-400 enabled:active:text-dark-neutral-600 dark:text-neutral-0 dark:enabled:hover:text-neutral-400 dark:enabled:active:text-neutral-600',
} as const;

const sizeTypes = {
	micro: 'py-0.5 px-2',
	slim: 'py-1 px-3',
	medium: 'py-2 px-4',
	large: 'py-3 px-6',
	custom: '',
} as const;

const iconSizes = {
	micro: 'w-3 h-3',
	slim: 'w-4 h-4',
	medium: 'w-5 h-5',
	large: 'w-5 h-5',
	custom: '',
} as const;

export default function Button({
	className,
	appearance = 'primary',
	iconAfter,
	iconBefore,
	isLoading,
	isDisabled,
	fullWidth,
	onBlur,
	onClick,
	onFocus,
	size = 'medium',
	children,
	href,
	iconButton,
	testId,
}: Props) {
	const currentFontSize = size === 'large' ? 'text-200' : 'text-100';
	const classes = [
		'items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150',
		iconButton ? 'text-0' : currentFontSize,
		isDisabled ? 'opacity-disabled cursor-not-allowed' : '',
		iconButton ? iconButtonAppearances[appearance] : appearanceTypes[appearance],
		iconButton ? '' : `${sizeTypes[size]} relative`,
		fullWidth ? 'w-full justify-center' : '',
		isLoading ? 'pointer-events-none text-transparent dark:text-transparent select-none' : '',
		className,
	].join(' ');

	const Children =
		(
			<>
				{!iconButton && isLoading && (
					<Spinner
						className='w-auto h-[80%] stroke-1 absolute top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2'
						size='custom'
						color={appearance === 'secondary' ? 'stroke-dark-neutral-0' : 'stroke-neutral-0'}
						testId='spinner'
					/>
				)}

				{!iconButton && iconBefore && (
					<span className={`${iconSizes[size]} ${size === 'micro' ? 'mr-1' : 'mr-2'}`}>
						{iconBefore}
					</span>
				)}

				<span
					className={iconButton ? 'inline-flex' : undefined}
				>
					{children}
				</span>

				{!iconButton && iconAfter && (
					<span className={`${iconSizes[size]} ${size === 'micro' ? 'ml-1' : 'ml-2'}`}>
						{iconAfter}
					</span>
				)}
			</>
		);

	if (href) {
		return (
			<Link
				className={classes}
				href={href}
				onClick={onClick}
				onBlur={onBlur}
				onFocus={onFocus}
				data-testid={testId}
			>
				{Children}
			</Link>
		);
	}

	return (
		<button
			className={classes}
			onClick={onClick}
			onBlur={onBlur}
			onFocus={onFocus}
			disabled={isDisabled}
			data-testid={testId}
		>
			{Children}
		</button>
	);
};