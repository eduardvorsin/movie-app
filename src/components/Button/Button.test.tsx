import { render, screen } from '@testing-library/react';
import Button, { Props } from './Button';
import React from 'react';
import userEvent from '@testing-library/user-event';

const appearances: NonNullable<Props['appearance']>[] = ['primary', 'secondary', 'warning', 'danger', 'success', 'discovery', 'ghost'];
const sizes: NonNullable<Props['size']>[] = ['micro', 'slim', 'medium', 'large'];

describe('Button tests', () => {
   it('renders correctly', () => {
      render(<Button>Button</Button>);

      expect(screen.getByRole<HTMLButtonElement>('button')).toBeInTheDocument();
   });

   it('when you click on the button, the mock function should work', async () => {
      const user = userEvent.setup();
      const mockFn = jest.fn();
      render(<Button onClick={mockFn}>Button</Button>);

      await user.click(screen.getByRole<HTMLButtonElement>('button'));

      expect(mockFn).toHaveBeenCalledTimes(1);
   });

   it('when the focus hits the button, the mock function should work', async () => {
      const user = userEvent.setup();
      const mockFn = jest.fn();
      render(<Button onFocus={mockFn}>Button</Button>);

      await user.tab();

      expect(mockFn).toHaveBeenCalledTimes(1);
   });

   it('if you lose focus from the button, the mock function should work', async () => {
      const user = userEvent.setup();
      const mockFn = jest.fn();
      render(<Button onBlur={mockFn}>Button</Button>);

      await user.tab();
      await user.click(document.body);

      expect(mockFn).toHaveBeenCalledTimes(1);
   });

   it('when isLoading is set to true, the spinner should appear', () => {
      render(<Button isLoading>Button</Button>);

      expect(screen.getByTestId<HTMLSpanElement>('spinner')).toBeInTheDocument();
   });

   it('with isDisabled set to true, the button should be inactive when clicked', async () => {
      const user = userEvent.setup();
      const mockFn = jest.fn();
      render(<Button isDisabled onClick={mockFn}>Button</Button>);

      await user.click(screen.getByRole<HTMLButtonElement>('button'));

      expect(mockFn).toHaveBeenCalledTimes(0);
   });

   it('is a basic snapshot', () => {
      render(<Button>Button</Button>);

      expect(screen.getByRole<HTMLButtonElement>('button')).toMatchSnapshot();
   });

   it.each(appearances)('is a snapshot with the "%s" appearance type', (appearance) => {
      render(<Button appearance={appearance}>Button</Button>);

      expect(screen.getByRole<HTMLButtonElement>('button')).toMatchSnapshot();
   });

   it.each(sizes)('is a snapshot with the "%s" size type', (size) => {
      render(<Button size={size}>Button</Button>);

      expect(screen.getByRole<HTMLButtonElement>('button')).toMatchSnapshot();
   });
});
