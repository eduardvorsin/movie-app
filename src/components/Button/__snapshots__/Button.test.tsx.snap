// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests is a basic snapshot 1`] = `
<button
  class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  bg-blue-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-blue-800 enabled:active:bg-blue-900 dark:bg-blue-400 dark:enabled:hover:bg-blue-300 dark:enabled:active:bg-blue-200 py-2 px-4 relative  "
>
  <span
    class="inline-flex items-center"
  >
    Button
  </span>
</button>
`;

exports[`Button tests is a snapshot with the "danger" appearance type 1`] = `
<button
  class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  bg-red-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-red-800 enabled:active:bg-red-900 dark:bg-red-400 dark:enabled:hover:bg-red-300 dark:enabled:active:bg-red-200 py-2 px-4 relative  "
>
  <span
    class="inline-flex items-center"
  >
    Button
  </span>
</button>
`;

exports[`Button tests is a snapshot with the "discovery" appearance type 1`] = `
<button
  class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  bg-purple-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-purple-800 enabled:active:bg-purple-900 dark:bg-purple-400 dark:enabled:hover:bg-purple-300 dark:enabled:active:bg-purple-200 py-2 px-4 relative  "
>
  <span
    class="inline-flex items-center"
  >
    Button
  </span>
</button>
`;

exports[`Button tests is a snapshot with the "ghost" appearance type 1`] = `
<button
  class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  text-dark-neutral-0 enabled:hover:text-dark-neutral-400 enabled:active:text-dark-neutral-600 dark:text-neutral-200 dark:enabled:hover:text-neutral-400 dark:enabled:active:text-neutral-600 py-2 px-4 relative  "
>
  <span
    class="inline-flex items-center"
  >
    Button
  </span>
</button>
`;

exports[`Button tests is a snapshot with the "large" size type 1`] = `
<button
  class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-200  bg-blue-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-blue-800 enabled:active:bg-blue-900 dark:bg-blue-400 dark:enabled:hover:bg-blue-300 dark:enabled:active:bg-blue-200 py-3 px-6 relative  "
>
  <span
    class="inline-flex items-center"
  >
    Button
  </span>
</button>
`;

exports[`Button tests is a snapshot with the "medium" size type 1`] = `
<button
  class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  bg-blue-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-blue-800 enabled:active:bg-blue-900 dark:bg-blue-400 dark:enabled:hover:bg-blue-300 dark:enabled:active:bg-blue-200 py-2 px-4 relative  "
>
  <span
    class="inline-flex items-center"
  >
    Button
  </span>
</button>
`;

exports[`Button tests is a snapshot with the "micro" size type 1`] = `
<button
  class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  bg-blue-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-blue-800 enabled:active:bg-blue-900 dark:bg-blue-400 dark:enabled:hover:bg-blue-300 dark:enabled:active:bg-blue-200 py-0.5 px-2 relative  "
>
  <span
    class="inline-flex items-center"
  >
    Button
  </span>
</button>
`;

exports[`Button tests is a snapshot with the "primary" appearance type 1`] = `
<button
  class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  bg-blue-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-blue-800 enabled:active:bg-blue-900 dark:bg-blue-400 dark:enabled:hover:bg-blue-300 dark:enabled:active:bg-blue-200 py-2 px-4 relative  "
>
  <span
    class="inline-flex items-center"
  >
    Button
  </span>
</button>
`;

exports[`Button tests is a snapshot with the "secondary" appearance type 1`] = `
<button
  class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  bg-neutral-300 text-dark-neutral-0 enabled:hover:bg-neutral-400 enabled:active:bg-neutral-500 dark:enabled:hover:bg-neutral-200 dark:enabled:active:bg-neutral-100 py-2 px-4 relative  "
>
  <span
    class="inline-flex items-center"
  >
    Button
  </span>
</button>
`;

exports[`Button tests is a snapshot with the "slim" size type 1`] = `
<button
  class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  bg-blue-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-blue-800 enabled:active:bg-blue-900 dark:bg-blue-400 dark:enabled:hover:bg-blue-300 dark:enabled:active:bg-blue-200 py-1 px-3 relative  "
>
  <span
    class="inline-flex items-center"
  >
    Button
  </span>
</button>
`;

exports[`Button tests is a snapshot with the "success" appearance type 1`] = `
<button
  class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  bg-green-700 text-neutral-0 dark:text-dark-neutral-0 enabled:hover:bg-green-800 enabled:active:bg-green-900 dark:bg-green-400 dark:enabled:hover:bg-green-300 dark:enabled:active:bg-green-200 py-2 px-4 relative  "
>
  <span
    class="inline-flex items-center"
  >
    Button
  </span>
</button>
`;

exports[`Button tests is a snapshot with the "warning" appearance type 1`] = `
<button
  class="items-center font-medium rounded-1 text-center cursor-pointer min-w[2.25rem] leading-2 inline-flex transition-colors duration-150 text-100  bg-yellow-500 text-neutral-1000 enabled:hover:bg-yellow-600 enabled:active:bg-yellow-700 dark:enabled:hover:bg-yellow-300 dark:enabled:active:bg-yellow-200 py-2 px-4 relative  "
>
  <span
    class="inline-flex items-center"
  >
    Button
  </span>
</button>
`;
